staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -160,14 +161,15 @@ type DeltaFIFO struct {
 	lock sync.RWMutex
 	cond sync.Cond
 
-	// We depend on the property that items in the set are in
-	// the queue and vice versa, and that all Deltas in this
-	// map have at least one Delta.
+	// `items` maps keys to Deltas and is the authority whether a key exists.
+	// `queue` maintains FIFO order of keys for consumption in Pop().
+	// Keys that exist in `queue` but not in `items` should be ignored.
```

Comment:
"should be ignored" is too broad. Presumably they have been deleted and the last thing in the queue is a deletion notification.

Reply:
I read the code more carefully. Even though in `Pop()` there's logic handling the case where a key exists in `queue` but not in `items`, with the current implementation, keys in `queue` and `items` are strictly 1:1 mapping. Keys are added to `items` and `queue` together and they are not appended to`queue` again if they already exist in `items` (inside `addIfNotPresent() and queueActionLocked()`).  Keys are removed from `items` and `queue` together in `Pop()`. The only place where keys are only removed from `items` but not `queue` is inside `queueActionLocked()` in a dead branch.

Did you notice anything that suggest otherwise? @lavalamp 

```
		id := f.queue[0]
		f.queue = f.queue[1:]
		if f.initialPopulationCount > 0 {
			f.initialPopulationCount--
		}
		item, ok := f.items[id]
		if !ok {
			// Item may have been deleted subsequently.
			continue
		}
		delete(f.items, id)
		err := process(item)
```

Reply:
I think this is a "should not happen" not a "should be ignored" :)

I'd just document that the code expects these to be 1:1.

Reply:
Yeah, I updated the doc to 

>We maintain the property that keys in the `items` and `queue` are strictly 1:1 mapping

Let me know what you think.